### PR TITLE
PROGRAMMES-5747 - Lx Promo

### PR DESCRIPTION
**Pull request by felipeparaujo**

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/ExternalApi/Morph/Service/LxPromoService.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+use BBC\ProgrammesPagesService\Domain\Entity\ProgrammeContainer;
+use BBC\ProgrammesPagesService\Domain\Exception\DataNotFetchedException;
+use GuzzleHttp\Promise\FulfilledPromise;
+use GuzzleHttp\Promise\PromiseInterface;
+use Psr\Log\LoggerInterface;
+use RMP\Translate\TranslateFactory;
+
+class LxPromoService
+{
+
```
Comment (votemike) [inline src/ExternalApi/Morph/Service/LxPromoService.php]: WHY IS THERE A BLANK LINE!? 😡

File: composer.json [hunk 2]
```diff
@@ -14,6 +14,8 @@
         { "type": "vcs", "url": "git@github.com:bbc/gel-iconography-assets.git" },
         { "type": "vcs", "url": "git@github.com:bbc/DoctrineExtensions.git" },
         { "type": "vcs", "url": "git@github.com:bbc/programmes-pages-service.git" },
+        { "type": "vcs", "url": "git@github.com:bbc/programmes-morph-library.git" },
+        { "type": "vcs", "url": "git@github.com:bbc/programmes-caching-library.git" },
```
Comment (votemike) [inline composer.json]: Why is this needed? Is this a dependency of programmes-morph-library?

File: src/ExternalApi/Morph/Service/LxPromoService.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    private $logger;
+
+    /** @var TranslateFactory */
+    private $translateFactory;
+
+    public function __construct(
+        LoggerInterface $logger,
+        TranslateFactory $translateFactory,
+        MorphClient $client,
+        string $env
```
Comment (votemike) [inline src/ExternalApi/Morph/Service/LxPromoService.php]: Do you think this env param could do with a bit more of a description?

File: src/ExternalApi/Morph/Service/LxPromoService.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    ) {
+        $this->env = $env;
+        $this->client = $client;
+        $this->logger = $logger;
+        $this->translateFactory = $translateFactory;
```
Comment (votemike) [inline src/ExternalApi/Morph/Service/LxPromoService.php]: Where's this used?

File: src/Controller/FindByPid/TlecController.php [hunk 5]
```diff
@@ -106,6 +109,35 @@ public function __invoke(
...
+
+        $promises = [
+            'recommendations' => $recommendationsPromise,
+            'relatedTopics' => $relatedTopicsPromise,
+            'supportingContentItems' => $electronService->fetchSupportingContentItemsForProgramme($programme),
+            'favouritesButton' => $favouritesButtonPromise,
+            'lxPromo' => $lxPromoPromise,
+        ];
+
+        $parameterPromises = $this->resolvePromises($promises);
```
Comment (BjornTwachtmann) [inline src/Controller/FindByPid/TlecController.php]: I don't think calling this variable promises when it now contains data is that clear.

File: composer.json (hunk 2 above)
Comment (felipeparaujo) [reply composer.json]: it's used both by morph-library and frontend now. We need to set the repo here because the package is not available from packagist. So everything that uses caching-libray, or that uses something that uses caching-library, need to define the repo.

File: src/ExternalApi/Morph/Service/LxPromoService.php [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    public function fetchByProgrammeContainer(ProgrammeContainer $programme): ?PromiseInterface
```
Comment (BjornTwachtmann) [inline src/ExternalApi/Morph/Service/LxPromoService.php]: Can this return null? It should always return a promise really.

File: composer.json (hunk 2 above)
Comment (votemike) [reply composer.json]: Where is it specifically required in this composer file?

Comment (felipeparaujo) [reply composer.json]: you're right, it's not necessary. i've removed it.
